### ISSUE TITLE
SavedStateHandle doesn't contain navigation arguments

### ISSUE THREAD

**Issue by grrigore** [InsertKoinIO/koin#1350]
I'm trying to inject `SavedStateHandle`. I am using Koin, Jetpack Compose and Navigation.
I've managed to make it work, but I think it's not the right instance (I don't get my values).

This is what I do so far:

Setting the argument:

```
private fun NavGraphBuilder.addDetails(
    navController: NavHostController,
    root: Screen
) {
    composable(route = LeafScreen.Details.createRoute(root),
        arguments = listOf(
            navArgument("showId") { type = NavType.LongType }
        )) {
        DetailsScreen(navigateUp = navController::navigateUp)
    }
}
```

Defining my ViewModel:
```
internal class DetailsViewModel(
    val handle: SavedStateHandle,
    private val getMovieDetailsUseCase: GetMovieDetailsUseCase
) : ViewModel() 
```

Defining my composable:
```
@Composable
fun DetailsScreen(
    navigateUp: () -> Unit
) {
    DetailsScreen(
        viewModel = getStateViewModel(),
        navigateUp = navigateUp,
    )
}
```

And my modules:
```
val uiDetailsModules = module {
    viewModelOf(::DetailsViewModel)
}
```

It runs but `showId` from `val showId = handle.get<Long>("showId")` is null.

**Comment by grrigore** [comment]
I have also tried using `viewModel { (handle: SavedStateHandle) -> DetailsViewModel(handle, get()) }`

**Comment by buhho-crtl** [comment]
Hello!
The same thing is happening to me, I think the case is related. According to the google changelog it should pick up the arguments via SavedStateHandle

https://developer.android.com/jetpack/androidx/releases/navigation#2.4.0

But this does not happen, here is my example.

Module:

 viewModel { (handle: SavedStateHandle) ->TestViewModel(savedStateHandle = handle) }

Fragment:

val viewModel: TestViewModelby stateViewModel()

ViewModel:

class TestViewModelby (savedStateHandle: SavedStateHandle)

Looking at the debug tab it seems that SavedStateHandle does not pick up the arguments from viewmodel

![image](https://user-images.githubusercontent.com/61876822/170481032-8f3aa5e8-6d9a-48bc-a25e-cd77a452c1df.png)

But if I look at the test arguments in the Fragment, it does have arguments.

![image](https://user-images.githubusercontent.com/61876822/170481561-f653dbe6-52c4-44ae-acf0-58dd5d158d3d.png)

If I try this using test by viewmodels() //androidx.fragment.app.viewModels 

It seems that if I get the arguments of the args from SavedStateHandle in the viewmodel, here is an example:

![image](https://user-images.githubusercontent.com/61876822/170482206-4fcc0253-17d8-41f0-861f-2051d04270c8.png)

I hope this helps to fix this :)

Thanks!

**Comment by rubenquadros** [comment]
Let's hope its available soon 😃
https://github.com/InsertKoinIO/koin/pull/1286

**Comment by buhho-crtl** [comment]
Thank you very much! :D

**Comment by Gwindor** [comment]
Looks like a good candidate for the patch. Any plans to release 3.2.1 with it?

**Comment by rubenquadros** [comment]
Please review the pr which confirm with 3.2 
https://github.com/InsertKoinIO/koin-compose/pull/3

**Comment by grrigore** [comment]
Still stuck with this one.

**Comment by qavan** [comment]
Workaround
```
private val EmptyBundle = Bundle()

fun Fragment.argumentsToBundleDefinition(): BundleDefinition = {
    if (arguments != null)
        arguments!!
    else
        EmptyBundle
}

inline fun <reified VM : ViewModel> Fragment.injectViewModel(
    qualifier: Qualifier? = null,
    noinline state: BundleDefinition = argumentsToBundleDefinition(),
    noinline owner: ViewModelStoreOwnerProducer = { this },
    noinline parameters: ParametersDefinition? = null,
): Lazy<VM> = stateViewModel(qualifier, state, owner, parameters)
```

**Comment by arnaudgiuliani** [comment]
Should be fixed in `koin-androidx-compose` `3.2.1`

PR https://github.com/InsertKoinIO/koin-compose/pull/3 has been merged

**Comment by szyman** [comment]
Thanks @arnaudgiuliani from `3.2.1` the `SavedStateHandle` started storing navigation arguments using `getViewModel` composable

**Comment by ibrahim-amrani-yassir** [comment]
@arnaudgiuliani i'm using version 3.5.6, using koinViewModel() in nav host but SaveStateHandle is null.

**Comment by arnaudgiuliani** [comment]
@ibrahim-amrani-yassir fixes are done in next 4.0 4.1 releases
